### PR TITLE
[Hungama] Add new extractor

### PR DESCRIPTION
**Pull request by awalgarg**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

New extractor for https://hungama.com which hosts a lot of Indian music. I've tested from Indian and Singaporean IP addresses, though there may be other geographical restrictions similar to Hotstar.

I may add support for playlists in a follow up PR.

### CODE REVIEW

**Comment by dstftw** [comment]
Read coding conventions and fix code.

**Comment by awalgarg** [comment]
Fixed everything I caught. Please let me know if I missed something.

File: youtube_dl/extractor/hungama.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        player_data_url = 'https://www.hungama.com/audio-player-data/track/%s?_country=IN' % video_id
+        player_data = self._download_json(player_data_url, video_id)
+        if len(player_data) != 1:
+            raise ExtractorError('Player data response invalid')
```
Comment (dstftw) [inline youtube_dl/extractor/hungama.py]: Remove.

File: youtube_dl/extractor/hungama.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        player_data = player_data[0]
+
+        title = player_data.get('song_name') or self._og_search_title(webpage)
+
+        track_data = self._download_json(player_data['file'], video_id)
+        if track_data.get('status') != 1:
+            raise ExtractorError('Failed to grab track data')
```
Comment (dstftw) [inline youtube_dl/extractor/hungama.py]: Remove.

File: youtube_dl/extractor/hungama.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            'ext': 'm3u8',
+            'title': 'Kitni Haseen Zindagi',
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        player_data_url = 'https://www.hungama.com/audio-player-data/track/%s?_country=IN' % video_id
```
Comment (dstftw) [inline youtube_dl/extractor/hungama.py]: Inline.

File: youtube_dl/extractor/hungama.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            raise ExtractorError('Player data response invalid')
+        player_data = player_data[0]
+
+        title = player_data.get('song_name') or self._og_search_title(webpage)
+
+        track_data = self._download_json(player_data['file'], video_id)
+        if track_data.get('status') != 1:
+            raise ExtractorError('Failed to grab track data')
+
+        url = track_data['response']['media_url']
```
Comment (dstftw) [inline youtube_dl/extractor/hungama.py]: Do not shadow existing variables.

File: youtube_dl/extractor/hungama.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import ExtractorError
+
+
+class HungamaIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)hungama\.com/song/(?P<title>[\w\d-]+)/(?P<id>[0-9]+)'
```
Comment (dstftw) [inline youtube_dl/extractor/hungama.py]: Remove unused groups. Relax regex.

**Comment by awalgarg** [comment]
Ah hold on, my bad. I should be using `_extract_m3u8_formats`. Fixed.

**Comment by awalgarg** [comment]
Addressed review comments

**Pull request closed without merging**